rs_zktrie/src/db.rs

```diff
@@ -1,13 +1,15 @@
 use crate::raw::ImplError;
 use std::collections::HashMap;
+use std::rc::Rc;
+
 pub trait ZktrieDatabase: Clone {
     fn put(&mut self, k: Vec<u8>, v: Vec<u8>) -> Result<(), ImplError>;
-    fn get(&self, k: &[u8]) -> Result<Vec<u8>, ImplError>;
+    fn get(&self, k: &[u8]) -> Result<Rc<[u8]>, ImplError>;
```

Comment:
I am afraid returning `Rc` would bring difficulilty to the possible concurrent improvment like we have recently merged.

Maybe simply return `&[u8]` is a more effecient choice.

Reply:
You cannot return a &[u8] here, the lifetime issue. I don't think it would cause any problem unless you are going to modify the entry in-place.

Reply:
I think the only reason to need a `Rc` is the value has to be kept after db has been dropped. There should be no such a case in our code.

Even the usage is valid, there is still some cost to apply a reference counting.

The lifetime issue is incuded by `RefCell` wrapping, which has been elimiated in the newest `main` branch. I would advice to merge the updates in `main` branch first, then try to apply a more efficient expression here.